core/domain/exp_services.py

```diff
@@ -1795,3 +1799,203 @@ def save_multi_exploration_math_rich_text_info_model(
 
     exp_models.ExplorationMathRichTextInfoModel.put_multi(
         exploration_math_rich_text_info_models)
+
+
+def generate_html_change_list_for_state(
+        state_name, new_state_dict, old_state_dict):
+    """Returns the change lists for all the html fields in a converted state
+    dict by comparing it with the corresponding old state dict.
+
+    Args:
+        state_name: str. The name of the state.
+        new_state_dict: dict. The dict representation of the nenw State object.
+        old_state_dict: dict. The dict representation of the old State object
+
+    Returns:
+        list(ExplorationChange). The generated change list.
+    """
+
+    change_lists = []
+    property_name_to_new_value_mapping = []
+    if old_state_dict['interaction']['customization_args'] != (
+            new_state_dict['interaction']['customization_args']):
+        property_name_to_new_value_mapping.append(
+            (
+                exp_domain.STATE_PROPERTY_INTERACTION_CUST_ARGS,
+                new_state_dict['interaction']['customization_args']))
+    if old_state_dict['content'] != (
+            new_state_dict['content']):
+        property_name_to_new_value_mapping.append(
+            (exp_domain.STATE_PROPERTY_CONTENT, new_state_dict['content']))
+
+    if old_state_dict['written_translations'] != (
+            new_state_dict['written_translations']):
+        property_name_to_new_value_mapping.append(
+            (
+                exp_domain.STATE_PROPERTY_WRITTEN_TRANSLATIONS,
+                new_state_dict['written_translations']))
+    if old_state_dict['interaction']['default_outcome'] != (
+            new_state_dict['interaction']['default_outcome']):
+        property_name_to_new_value_mapping.append(
+            (
+                exp_domain.STATE_PROPERTY_INTERACTION_DEFAULT_OUTCOME,
+                new_state_dict['interaction']['default_outcome']))
+    if old_state_dict['interaction']['hints'] != (
+            new_state_dict['interaction']['hints']):
+        property_name_to_new_value_mapping.append(
+            (
+                exp_domain.STATE_PROPERTY_INTERACTION_HINTS,
+                new_state_dict['interaction']['hints']))
+    if old_state_dict['interaction']['solution'] != (
+            new_state_dict['interaction']['solution']):
+        property_name_to_new_value_mapping.append(
+            (
+                exp_domain.STATE_PROPERTY_INTERACTION_SOLUTION,
+                new_state_dict['interaction']['solution']))
+    if old_state_dict['interaction']['answer_groups'] != (
```

Comment:
This all `if` looks like kinda repeated, maybe we can find a better way to do it i.e, more readable and less code? (This is not that important though, feel free to leave it as-is if you won't find a better way!)

Reply:
I actually couldn't find a simpler way, pls do suggest if you find a simpler way